src/Compilers/CSharp/Portable/Lowering/LambdaRewriter/LambdaRewriter.Analysis.Tree.cs

```diff
@@ -0,0 +1,505 @@
+                    return base.VisitParameter(node);
+                }
+
+                public override BoundNode VisitLocal(BoundLocal node)
+                {
+                    AddIfCaptured(node.LocalSymbol);
+                    return base.VisitLocal(node);
+                }
+
+                public override BoundNode VisitBaseReference(BoundBaseReference node)
+                {
+                    AddIfCaptured(_analysis._topLevelMethod.ThisParameter);
+                    return base.VisitBaseReference(node);
+                }
+
+                public override BoundNode VisitThisReference(BoundThisReference node)
+                {
+                    var thisParam = _analysis._topLevelMethod.ThisParameter;
+                    if (thisParam != null)
+                    {
+                        AddIfCaptured(thisParam);
+                    }
+                    else
+                    {
+                        // This can occur in a delegate creation expression because the method group
+                        // in the argument can have a "this" receiver even when "this"
+                        // is not captured because a static method is selected.  But we do preserve
+                        // the method group and its receiver in the bound tree.
+                        // No need to capture "this" in such case.
+
+                        // TODO: Why don't we drop "this" while lowering if method is static? 
+                        //       Actually, considering that method group expression does not evaluate to a particular value 
+                        //       why do we have it in the lowered tree at all?
+                    }
+
+                    return base.VisitThisReference(node);
+                }
+
+                private BoundNode VisitClosure(MethodSymbol closureSymbol, BoundBlock body)
+                {
+                    Debug.Assert((object)closureSymbol != null);
+
+                    // Closure is declared (lives) in the parent scope, but its
+                    // variables are in a nested scope
+                    var closure = new Closure(closureSymbol);
+                    _currentScope.Closures.Add(closure);
+
+                    var oldClosure = _currentClosure;
+                    _currentClosure = closure;
+
+                    var oldScope = _currentScope;
+                    _currentScope = CreateNestedScope(body, _currentScope, _currentClosure);
+
+                    BoundNode result;
+                    if (!_inExpressionTree)
+                    {
+                        // For the purposes of scoping, parameters live in the same scope as the
+                        // closure block
+                        DeclareLocals(_currentScope, closureSymbol.Parameters);
+                        result = VisitBlock(body);
+                    }
+                    else
+                    {
+                        result = base.VisitBlock(body);
```

Comment:
Why do we use `base.` here but not in the call above? Is our VisitBlock method not setup to handle expression trees? What happens if in the future we add blocks in expressin trees? Guessing that will break VisitBlock.